### PR TITLE
simplify naming

### PR DESCRIPTION
**Pull request by faddat**
## Description

Binary names and paths were too long.

This PR:

* interchain-security-cd => consumer
* interchain-security-cdd => democracy
* interchain-security-pd => provider

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if API or client breaking change
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [x] Reviewed "Files changed" and left comments if necessary 
* [x] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed all author checklist items have been addressed 
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

**Comment by jtremback** [comment]
This just changes the binary names? I like the idea

**Comment by faddat** [comment]
yep.  I just need to clean it up and such, should be able to get to it tomorrow. 

It is inspired by typing:

interchain-security-pd 

too many times

:)

**Comment by jtremback** [comment]
I'm getting test failures on this even after re-running. Can you check that the e2e tests all have the updated binary names?

**Comment by faddat** [comment]
ah that's why it is still on draft.

**Comment by faddat** [comment]
hmmmm I thought i had this working....

**Comment by faddat** [comment]
@jtremback please note that we could debug this directly on github, if we changed the automated tests to use the --verbose flag as seen in #867

**Comment by faddat** [comment]
For reference, 5/5 vietnamese undersea cables have some kind of fault right now. 

Thus, i am still downloading docker containers.

**Comment by jtremback** [comment]
LGTM but looks like some merge conflicts snuck in

**Comment by smarshall-spitzbart** [comment]
Set this to draft, but once the merge conflicts are fixed I'm happy to review

**Comment by faddat** [comment]
hey really sorry guys!  

this one has been a while 1 moment while I do the conflict resolution dance.

**Comment by smarshall-spitzbart** [comment]
There's an sdk pattern that binary names have a `d` appended, was there reasoning around avoiding that pattern?

Ex: `providerd` or `provider-d`

**Comment by faddat** [comment]
@smarshall-spitzbart it's broken inconssitently throughout the ecosystem.  I don't mind appending d's to the names though, and prefer consistency.

**Comment by smarshall-spitzbart** [comment]
If we're taking the time to change/review the binary names I'd prefer we do it the proper way with the d appended, thanks @faddat

**Comment by faddat** [comment]
Gonna add the d now...

File: README.md
Comment (smarshall-spitzbart) [inline README.md:35]:
```suggestion
providerd
```

File: README.md
Comment (smarshall-spitzbart) [inline README.md:37]:
```suggestion
consumerd
```

File: app/democracy/app.go
Comment (smarshall-spitzbart) [inline app/democracy/app.go:122]:
```suggestion
	AppName              = "democracyd"
```

File: docs/docs/features/key-assignment.md
Comment (smarshall-spitzbart) [inline docs/docs/features/key-assignment.md:34]: Keep `d`s in this file?

**Review by smarshall-spitzbart — COMMENTED** [review]
Not sure that changing "democracy consumer" to "democracy" is beneficial or makes things clearer. 

Making binary names shorter is fine (that's what this PR was originally intended to do). 

Aside: There's plenty of open issues which describe refactors that the core team would find to be useful, see https://github.com/cosmos/interchain-security/issues?q=is%3Aopen+is%3Aissue+label%3Arefactoring

File: docs/docs/features/key-assignment.md
Comment (faddat) [reply docs/docs/features/key-assignment.md:34]: totally.  sorry bout that.

**Comment by faddat** [comment]
> Not sure that changing "democracy consumer" to "democracy" is beneficial or makes things clearer.



yeah the idea was def to make binary names shorter, any other name changes, were done for the sake of consistency is there a specific instance where this is an issue?  

Happy to address

**Comment by faddat** [comment]
> https://github.com/cosmos/interchain-security/issues?q=is%3Aopen+is%3Aissue+label%3Arefactoring

🖖 

thank you

File: testutil/ibc_testing/specific_setup.go
Comment (faddat) [inline testutil/ibc_testing/specific_setup.go:17]: was this what you were unsure of @smarshall-spitzbart ?

**Comment by jtremback** [comment]
Merge conflicts snuck in again, can merge once it's fixed though

**Comment by faddat** [comment]
@jtremback hey sorry about the delay here.  Baby gap sir.

**Comment by faddat** [comment]
I've merged main, hopefully this fixes the cometmocktest.

File: docs/docs/features/key-assignment.md
Comment (mpoke) [inline docs/docs/features/key-assignment.md:66]: This should be `consumerd`.

**Review by mpoke — APPROVED** [review]
LGTM. Just one nit below.

**Comment by mpoke** [comment]
@faddat Could you please also add  the compiler/linker flags (`ldflags`) during the build call so that we can also close https://github.com/cosmos/interchain-security/issues/1035? We decided against goreleaser for ICS, but it would be useful to populate those fields for the version CLI command.

File: Dockerfile
Comment (mpoke) [inline Dockerfile:52]: Could you also change sovereign to standalone? Calling a non-ICS chain sovereign, entails that consumer chains don't have sovereignty, which is not true.

Comment (faddat) [reply Dockerfile:52]: yeah I'm good with this change sir

**Comment by mpoke** [comment]
Closing this as it is out of sync with main. It's easier to redo the PR than to rebase.

**Pull request closed without merging**
